### PR TITLE
Removed MEDIA_IMAGE_PHOTON References from codebase

### LINKED ISSUE #27446: Janitorial: remove MEDIA_IMAGE_PHOTON from codebase
I recently updated the media thumbnails to not use Photon in https://github.com/Automattic/wp-calypso/pull/27412

There was a review comment by @ramonjd which I'll raise as as a separate issue to work on so I can merge the PR as it is:

> If we're no longer using this constant for images, perhaps it's a good chance to cull it. For example, the default thumbnailType for <ListItemImage /> is still MEDIA_IMAGE_PHOTON.
> 
> I'm not sure however how photon relates to videos, but it looks like we're passing down thumbnailType to all <ListItem /> children, so on the face of it, it appears we can change the default type in all child components.
>
> It might be worth doing should we, for any reason, wish to prevent MEDIA_IMAGE_PHOTON from being the default. For example, if the child components don't receive a valid props.thumbnailType.
> 
> If you do decide to change the default type, then we can also get rid of the corresponding tests.

### PR DESCRIPTION
**Pull request by rfaile313**
#### Changes proposed in this Pull Request

Fixes #27446

---

#### Testing instructions

Not entirely sure how to test. Removing the references and compiling doesn't seem to have any impact on a local compile. 

screenshot

---

Let me know if you'd like me to change/look at anything else 🙏

### CODE REVIEW

**Comment by ramonjd** [comment]
Thanks @rfaile313 

This works for me locally too, and aside from the linting/unit test CI failures, I don't see any probs.

I wasn't originally sure whether we still needed `MEDIA_IMAGE_PHOTON` at all, assumed we didn't based on the title of #27446 (_Janitorial: remove MEDIA_IMAGE_PHOTON from codebase_)

I don't have the full low down on this 

cc @sgomes and @johngodley who might be able to confirm.

**Comment by johngodley** [comment]
I can't really provide much insight, however from my brief experience the things to test are:

- Images on a public site
- Images from Google Photos and Pexels - these don't support `resize=x`
- Images on a private site

Everything seems to check out when I try the above, and I think the unit tests cover this.

**Comment by rfaile313** [comment]
Noice. Merge?

**Comment by johngodley** [comment]
Pulling in the testing instructions from the original PR:

* Make sure media src URLs are files.wordpress.com ones (not photon like iX.wp.com)
* Make sure this works for WordPress.com simple public, private, Atomic and Jetpack sites

I haven't been able to test on Jetpack or atomic, so someone should try that.

Otherwise, just the unit tests need fixing and it should be good to go.

**Comment by rfaile313** [comment]
I'll perform the tests:

* Make sure media src URLs are files.wordpress.com ones (not photon like iX.wp.com)
* Make sure this works for WordPress.com simple public, private, Atomic and Jetpack sites

Not sure about the unit tests if someone wouldn't mind helping :-)

**Comment by ramonjd** [comment]
> Not sure about the unit tests if someone wouldn't mind helping 

Since we're not using photon for the image list item, you can replace `getPhotonUrl()` with `getResizedUrl()` in the [corresponding test](https://github.com/Automattic/wp-calypso/blob/ec7d2ec/client/my-sites/media-library/test/list-item-image.jsx#L49). This means you can also remove the `getPhotonUrl()` [method](https://github.com/Automattic/wp-calypso/blob/ec7d2ec/client/my-sites/media-library/test/list-item-image.jsx#L32). 

The tests should pass :) You can run them locally with the following command:

```
npm run test-client client/my-sites/media-library/test  

```

**Comment by sgomes** [comment]
@ramonjd Sorry, can't offer much insight on whether removing `MEDIA_IMAGE_PHOTON` will break anything 😕

My work in images in Calypso was focused on reducing image dimensions and file sizes across the board, where possible.

**Comment by rfaile313** [comment]
* Make sure media src URLs are files.wordpress.com ones (not photon like iX.wp.com)

* Make sure this works for WordPress.com simple public, private, Atomic and Jetpack sites

- Simple screenshot
- AT screenshot
- Jetpack - screenshot

>npm run test-client client/my-sites/media-library/test

```
Test Suites: 7 passed, 7 total
Tests:       47 passed, 47 total
Snapshots:   0 total
Time:        19.784s
Ran all test suites matching /client\/my-sites\/media-library\/test/i.
```

File: client/my-sites/media-library/test/list-item-image.jsx [hunk 1]
```diff
@@ -29,7 +29,8 @@ describe( 'MediaLibraryListItem image', () => {
 		}
 	} );
 
-	const getPhotonUrl = () => photon( fixtures.media[ 0 ].URL, { resize: `${ WIDTH },${ WIDTH }` } );
+	const getResizedUrl = () =>
```
Comment (ramonjd) [inline client/my-sites/media-library/test/list-item-image.jsx]: Why is this method duplicated? I'm pretty sure just the original `getResizedUrl` needs to be there. It's the reason behind the linting failure.

Comment (ramonjd) [reply client/my-sites/media-library/test/list-item-image.jsx]: Speaking of linting, there are a couple of [errors](https://circleci.com/gh/Automattic/wp-calypso/204662?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

```
line 131, col 4, Error - Visible, non-interactive elements with click handlers must have at least one keyboard listener. (jsx-a11y/click-events-have-key-events)

line 131, col 4, Error - Static HTML elements with event handlers require a role. (jsx-a11y/no-static-element-interactions)

```

All these accessibility issues need to be addressed at some point, but I don't think it's in scope for this PR.

You can disable these warning by following this [example](https://github.com/Automattic/wp-calypso/blob/87ea7231a319c3b9a06fb021308324ed66b4db23/client/components/title-format-editor/token.jsx#L6).

Hope that helps!

Comment (rfaile313) [reply client/my-sites/media-library/test/list-item-image.jsx]: I looked at this today but I'm not sure I understand. I'm only seeing a single definition for getResizedUrl() then it's called a few times later. I'm also interested in solving the linting issues but not entirely sure how.

Sorry, I know this isn't all that important. Just unsure how to proceed 😢

Comment (ramonjd) [reply client/my-sites/media-library/test/list-item-image.jsx]: No worries @rfaile313 

I'll take a quick look and push a commit if that's okay to you. I'll write a couple of notes to explain the changes where appropriate.

Comment (ramonjd) [reply client/my-sites/media-library/test/list-item-image.jsx]: Just to clarify, here's what I'm seeing in the [test file](https://github.com/Automattic/wp-calypso/blob/e342f7c0f08ffdd3a2aae8e1b114a900f32c5cf2/client/my-sites/media-library/test/list-item-image.jsx#L32).

```
	const getResizedUrl = () =>
		photon( fixtures.media[ 0 ].URL, { resize: `${ WIDTH },${ WIDTH }` } );
	const getResizedUrl = () =>
		resize( fixtures.media[ 0 ].URL, { resize: `${ WIDTH },${ WIDTH }` } );
``

There are two functions with the same name. The one calling `photon()` has been added in e342f7c0f08ffdd3a2aae8e1b114a900f32c5cf2. We'll ditch it.

As for the other linting errors, wrapping the rendered element in the `eslint-disable` comments will ensure that the linter doesn't complain. For example, to address the linting errors in [my-sites/media-library/list-item.jsx](https://github.com/Automattic/wp-calypso/blob/e342f7c0f08ffdd3a2aae8e1b114a900f32c5cf2/client/my-sites/media-library/list-item.jsx#L131):


	/* eslint-disable jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */
	<div className={ classes } style={ style } onClick={ this.clickItem } { ...props }>
		...
	</div>
	/* eslint-enable jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */

I'll rebase and make these changes. 👍

**Comment by ramonjd** [comment]
@rfaile313 It might be worth testing again, since I rebased. Just to make sure :)

Should be fine.

Also, if you want to check that your changes pass the linting rules, you can run eslint locally by executing the following command:

```
npm run eslint-branch   
```

You also can set eslint to run automatically when you commit changes. Here's more info on that: https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#automatically-run-eslint-against-your-changesets

File: client/my-sites/media-library/test/list-item-image.jsx (hunk 1 above)
Comment (rfaile313) [reply client/my-sites/media-library/test/list-item-image.jsx]: Hey Thanks! 

Am I crazy? 

In my local commit and in the github PR here 

>const getResizedUrl = () =>
		photon( fixtures.media[ 0 ].URL, { resize: `${ WIDTH },${ WIDTH }` } );
	const getResizedUrl = () =>
		resize( fixtures.media[ 0 ].URL, { resize: `${ WIDTH },${ WIDTH }` } );

I had removed the the first instance (calling photon) and I wasn't seeing it in the code. I must be losing it 😅

Thanks for your patience with this. I'll make these changes as well on my local branch and commit but if you have it from here that's cool too. Appreciate your time with this. A12 devs rock!

**Comment by rfaile313** [comment]
Meant to come back to this. LGTM 👍 👍

**Pull request merged**